<?xml version='1.0' encoding='UTF-8' standalone='no'?>
<doxygen xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="compound.xsd" version="1.8.11">
  <compounddef id="classnode__manager__client_1_1exceptions_1_1ApiTypeError" kind="class" language="Python" prot="public">
    <compoundname>node_manager_client::exceptions::ApiTypeError</compoundname>
    <basecompoundref refid="classnode__manager__client_1_1exceptions_1_1OpenApiException" prot="public" virt="non-virtual">node_manager_client.exceptions.OpenApiException</basecompoundref>
    <basecompoundref prot="public" virt="non-virtual">TypeError</basecompoundref>
    <basecompoundref refid="classnode__manager__client_1_1exceptions_1_1OpenApiException" prot="public" virt="non-virtual">node_manager_client.exceptions.OpenApiException</basecompoundref>
    <basecompoundref prot="public" virt="non-virtual">TypeError</basecompoundref>
      <sectiondef kind="public-attrib">
      <memberdef kind="variable" id="classnode__manager__client_1_1exceptions_1_1ApiTypeError_1a74194e24321581c6030151d5c65a94a6" prot="public" static="no" mutable="no">
        <type></type>
        <definition>node_manager_client.exceptions.ApiTypeError::path_to_item</definition>
        <argsstring></argsstring>
        <name>path_to_item</name>
        <briefdescription>
        </briefdescription>
        <detaileddescription>
        </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="OpenAPI/tests/editable_copies/node_manager_client/node_manager_client/exceptions.py" line="40" column="1" bodyfile="OpenAPI/tests/editable_copies/node_manager_client/node_manager_client/exceptions.py" bodystart="40" bodyend="-1"/>
      </memberdef>
      <memberdef kind="variable" id="classnode__manager__client_1_1exceptions_1_1ApiTypeError_1a67cb07c30835af5b8831d3d56d424181" prot="public" static="no" mutable="no">
        <type></type>
        <definition>node_manager_client.exceptions.ApiTypeError::valid_classes</definition>
        <argsstring></argsstring>
        <name>valid_classes</name>
        <briefdescription>
        </briefdescription>
        <detaileddescription>
        </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="OpenAPI/tests/editable_copies/node_manager_client/node_manager_client/exceptions.py" line="41" column="1" bodyfile="OpenAPI/tests/editable_copies/node_manager_client/node_manager_client/exceptions.py" bodystart="41" bodyend="-1"/>
      </memberdef>
      <memberdef kind="variable" id="classnode__manager__client_1_1exceptions_1_1ApiTypeError_1ab44f34d8614ec3712ec2341f5844858e" prot="public" static="no" mutable="no">
        <type></type>
        <definition>node_manager_client.exceptions.ApiTypeError::key_type</definition>
        <argsstring></argsstring>
        <name>key_type</name>
        <briefdescription>
        </briefdescription>
        <detaileddescription>
        </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="OpenAPI/tests/editable_copies/node_manager_client/node_manager_client/exceptions.py" line="42" column="1" bodyfile="OpenAPI/tests/editable_copies/node_manager_client/node_manager_client/exceptions.py" bodystart="42" bodyend="-1"/>
      </memberdef>
      </sectiondef>
      <sectiondef kind="public-func">
      <memberdef kind="function" id="classnode__manager__client_1_1exceptions_1_1ApiTypeError_1a49b57dc8237788027798e0e5e278e0ba" prot="public" static="no" const="no" explicit="no" inline="no" virt="non-virtual">
        <type>def</type>
        <definition>def node_manager_client.exceptions.ApiTypeError.__init__</definition>
        <argsstring>(self, msg, path_to_item=None, valid_classes=None, key_type=None)</argsstring>
        <name>__init__</name>
        <param>
          <type>self</type>
          <defname>self</defname>
        </param>
        <param>
          <type>msg</type>
          <defname>msg</defname>
        </param>
        <param>
          <type>path_to_item</type>
          <defname>path_to_item</defname>
          <defval>None</defval>
        </param>
        <param>
          <type>valid_classes</type>
          <defname>valid_classes</defname>
          <defval>None</defval>
        </param>
        <param>
          <type>key_type</type>
          <defname>key_type</defname>
          <defval>None</defval>
        </param>
        <briefdescription>
        </briefdescription>
        <detaileddescription>
<para><verbatim>Raises an exception for TypeErrors

Args:
    msg (str): the exception message

Keyword Args:
    path_to_item (list): a list of keys an indices to get to the
                 current_item
                 None if unset
    valid_classes (tuple): the primitive classes that current item
                   should be an instance of
                   None if unset
    key_type (bool): False if our value is a value in a dict
             True if it is a key in a dict
             False if our item is an item in a list
             None if unset
</verbatim> </para>        </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="OpenAPI/tests/editable_copies/node_manager_client/node_manager_client/exceptions.py" line="21" column="1" bodyfile="OpenAPI/tests/editable_copies/node_manager_client/node_manager_client/exceptions.py" bodystart="22" bodyend="48"/>
      </memberdef>
      <memberdef kind="function" id="classnode__manager__client_1_1exceptions_1_1ApiTypeError_1a49b57dc8237788027798e0e5e278e0ba" prot="public" static="no" const="no" explicit="no" inline="no" virt="non-virtual">
        <type>def</type>
        <definition>def node_manager_client.exceptions.ApiTypeError.__init__</definition>
        <argsstring>(self, msg, path_to_item=None, valid_classes=None, key_type=None)</argsstring>
        <name>__init__</name>
        <param>
          <type>self</type>
          <defname>self</defname>
        </param>
        <param>
          <type>msg</type>
          <defname>msg</defname>
        </param>
        <param>
          <type>path_to_item</type>
          <defname>path_to_item</defname>
          <defval>None</defval>
        </param>
        <param>
          <type>valid_classes</type>
          <defname>valid_classes</defname>
          <defval>None</defval>
        </param>
        <param>
          <type>key_type</type>
          <defname>key_type</defname>
          <defval>None</defval>
        </param>
        <briefdescription>
        </briefdescription>
        <detaileddescription>
<para><verbatim>Raises an exception for TypeErrors

Args:
    msg (str): the exception message

Keyword Args:
    path_to_item (list): a list of keys an indices to get to the
                 current_item
                 None if unset
    valid_classes (tuple): the primitive classes that current item
                   should be an instance of
                   None if unset
    key_type (bool): False if our value is a value in a dict
             True if it is a key in a dict
             False if our item is an item in a list
             None if unset
</verbatim> </para>        </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="OpenAPI/tests/node_manager_client/node_manager_client/exceptions.py" line="21" column="1" bodyfile="OpenAPI/tests/node_manager_client/node_manager_client/exceptions.py" bodystart="22" bodyend="48"/>
      </memberdef>
      </sectiondef>
    <briefdescription>
    </briefdescription>
    <detaileddescription>
    </detaileddescription>
    <inheritancegraph>
      <node id="2040">
        <label>node_manager_client.exceptions.OpenApiException</label>
        <link refid="classnode__manager__client_1_1exceptions_1_1OpenApiException"/>
        <childnode refid="2041" relation="public-inheritance">
        </childnode>
        <childnode refid="2041" relation="public-inheritance">
        </childnode>
      </node>
      <node id="2039">
        <label>node_manager_client.exceptions.ApiTypeError</label>
        <link refid="classnode__manager__client_1_1exceptions_1_1ApiTypeError"/>
        <childnode refid="2040" relation="public-inheritance">
        </childnode>
        <childnode refid="2042" relation="public-inheritance">
        </childnode>
        <childnode refid="2040" relation="public-inheritance">
        </childnode>
        <childnode refid="2042" relation="public-inheritance">
        </childnode>
      </node>
      <node id="2041">
        <label>Exception</label>
      </node>
      <node id="2042">
        <label>TypeError</label>
      </node>
    </inheritancegraph>
    <collaborationgraph>
      <node id="2044">
        <label>node_manager_client.exceptions.OpenApiException</label>
        <link refid="classnode__manager__client_1_1exceptions_1_1OpenApiException"/>
        <childnode refid="2045" relation="public-inheritance">
        </childnode>
        <childnode refid="2045" relation="public-inheritance">
        </childnode>
      </node>
      <node id="2043">
        <label>node_manager_client.exceptions.ApiTypeError</label>
        <link refid="classnode__manager__client_1_1exceptions_1_1ApiTypeError"/>
        <childnode refid="2044" relation="public-inheritance">
        </childnode>
        <childnode refid="2046" relation="public-inheritance">
        </childnode>
        <childnode refid="2044" relation="public-inheritance">
        </childnode>
        <childnode refid="2046" relation="public-inheritance">
        </childnode>
      </node>
      <node id="2045">
        <label>Exception</label>
      </node>
      <node id="2046">
        <label>TypeError</label>
      </node>
    </collaborationgraph>
    <location file="OpenAPI/tests/editable_copies/node_manager_client/node_manager_client/exceptions.py" line="20" column="1" bodyfile="OpenAPI/tests/editable_copies/node_manager_client/node_manager_client/exceptions.py" bodystart="20" bodyend="49"/>
    <listofallmembers>
      <member refid="classnode__manager__client_1_1exceptions_1_1ApiTypeError_1a49b57dc8237788027798e0e5e278e0ba" prot="public" virt="non-virtual"><scope>node_manager_client::exceptions::ApiTypeError</scope><name>__init__</name></member>
      <member refid="classnode__manager__client_1_1exceptions_1_1ApiTypeError_1a49b57dc8237788027798e0e5e278e0ba" prot="public" virt="non-virtual"><scope>node_manager_client::exceptions::ApiTypeError</scope><name>__init__</name></member>
      <member refid="classnode__manager__client_1_1exceptions_1_1ApiTypeError_1ab44f34d8614ec3712ec2341f5844858e" prot="public" virt="non-virtual"><scope>node_manager_client::exceptions::ApiTypeError</scope><name>key_type</name></member>
      <member refid="classnode__manager__client_1_1exceptions_1_1ApiTypeError_1a74194e24321581c6030151d5c65a94a6" prot="public" virt="non-virtual"><scope>node_manager_client::exceptions::ApiTypeError</scope><name>path_to_item</name></member>
      <member refid="classnode__manager__client_1_1exceptions_1_1ApiTypeError_1a67cb07c30835af5b8831d3d56d424181" prot="public" virt="non-virtual"><scope>node_manager_client::exceptions::ApiTypeError</scope><name>valid_classes</name></member>
    </listofallmembers>
  </compounddef>
</doxygen>
